### ISSUE TITLE
Fee Schedule claims to be a duplicate when reverting to a previous schedule

### ISSUE THREAD

**Issue by Krakaw** [big-neon/bn-api#633]
Create a fee schedule:
```
Minimum Price: 1     Fee: 0.5
Minimum Price: 10   Fee: 1.5
```
Then delete the second fee.
Then recreate it and it will fail as a duplicate.

**Comment by Timo614** [comment]
How does the deletion above work? Just assigning a new fee schedule to the organization with the range not present in the new version?

I tried to do this via the command line but didn't see a way to trigger an error. Had to use integers as the backend doesn't take floating point numbers so converted the dollars into cents.

Couldn't find any other way in the system to update these so think I have the right endpoint / flow here for triggering this. Let me know though if I'm incorrectly doing something with the reproduction steps.

Looking at the db the two unique indexes are:

```
CREATE UNIQUE INDEX index_fee_schedules_name ON fee_schedules(name, version);
CREATE UNIQUE INDEX index_fee_schedule_ranges_fee_schedule_id_min_price ON fee_schedule_ranges(fee_schedule_id, min_price)
```

So to trigger an error here we'd either need to try and create a brand new fee schedule with the same name and version as the other (perhaps a race condition if we're triggering two updates in tandem?) or we'd need to pass in a dupe value for the range for the given fee schedule. I think the latter is less likely to happen in practice as I couldn't find a way to update an existing fee schedule from code so you'd be in the add fee schedules logic which would have a brand new fee schedule (so would only trigger if the ranges array had dupes in it).

Create initial fee schedule:

```
curl http://localhost:9000/organizations/39446110-6f19-4f81-8002-b526b74beb2e/fee_schedule --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIwMGI0MThiMC0yYzg0LTQ5NjktODliZi0xNTQ4ZWRhNzVhMmEiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTQzODc3NTQzfQ.JigOQwrkeXCwmMNPswQjMA0zqt8hOt9QCrXN4D0-8Ck" --request POST --data '{"name": "Organization Fee Schedule", "ranges": [{"min_price": 100, "fee_in_cents": 50}, {"min_price": 1000, "fee_in_cents": 150}]}' --header "Content-Type: application/json" -i
{"id":"02847fb9-bb0b-4ec6-81f9-2da091df484c","name":"Organization Fee Schedule","version":0,"created_at":"2018-12-03T22:38:03.179941","ranges":[{"id":"bfc4c7bc-bf18-43fa-9e5e-0cfb9febc237","fee_schedule_id":"02847fb9-bb0b-4ec6-81f9-2da091df484c","min_price":100,"fee_in_cents":50,"created_at":"2018-12-03T22:38:03.179941","updated_at":"2018-12-03T22:38:03.179941"},{"id":"bcc6ac84-63f2-4224-b2d8-5bfd387bce96","fee_schedule_id":"02847fb9-bb0b-4ec6-81f9-2da091df484c","min_price":1000,"fee_in_cents":150,"created_at":"2018-12-03T22:38:03.179941","updated_at":"2018-12-03T22:38:03.179941"}]}

select * from fee_schedules where name = 'Organization Fee Schedule';
                id                  |           name            | version |         created_at         |         updated_at         
--------------------------------------+---------------------------+---------+----------------------------+----------------------------
02847fb9-bb0b-4ec6-81f9-2da091df484c | Organization Fee Schedule |       0 | 2018-12-03 22:38:03.179941 | 2018-12-03 22:38:03.179941

select fsr.* from fee_schedule_ranges fsr join fee_schedules f on f.id = fsr.fee_schedule_id where f.name = 'Organization Fee Schedule' and f.version = 0;
                 id                  |           fee_schedule_id            | min_price | fee_in_cents |         created_at         |         updated_at         
--------------------------------------+--------------------------------------+-----------+--------------+----------------------------+----------------------------
bfc4c7bc-bf18-43fa-9e5e-0cfb9febc237 | 02847fb9-bb0b-4ec6-81f9-2da091df484c |       100 |           50 | 2018-12-03 22:38:03.179941 | 2018-12-03 22:38:03.179941
bcc6ac84-63f2-4224-b2d8-5bfd387bce96 | 02847fb9-bb0b-4ec6-81f9-2da091df484c |      1000 |          150 | 2018-12-03 22:38:03.179941 | 2018-12-03 22:38:03.179941
```

Create new fee schedule removing second range:

```
curl http://localhost:9000/organizations/39446110-6f19-4f81-8002-b526b74beb2e/fee_schedule --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIwMGI0MThiMC0yYzg0LTQ5NjktODliZi0xNTQ4ZWRhNzVhMmEiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTQzODc3NTQzfQ.JigOQwrkeXCwmMNPswQjMA0zqt8hOt9QCrXN4D0-8Ck" --request POST --data '{"name": "Organization Fee Schedule", "ranges": [{"min_price": 100, "fee_in_cents": 50}]}' --header "Content-Type: application/json" -i
{"id":"4d23a63b-7c61-46af-b53d-9bb5d2650fab","name":"Organization Fee Schedule","version":1,"created_at":"2018-12-03T22:39:10.466736","ranges":[{"id":"0de49c41-7866-4e77-b3f9-f857d6fad56a","fee_schedule_id":"4d23a63b-7c61-46af-b53d-9bb5d2650fab","min_price":100,"fee_in_cents":50,"created_at":"2018-12-03T22:39:10.466736","updated_at":"2018-12-03T22:39:10.466736"}]}

select * from fee_schedules where name = 'Organization Fee Schedule';
                id                  |           name            | version |         created_at         |         updated_at         
--------------------------------------+---------------------------+---------+----------------------------+----------------------------
02847fb9-bb0b-4ec6-81f9-2da091df484c | Organization Fee Schedule |       0 | 2018-12-03 22:38:03.179941 | 2018-12-03 22:38:03.179941
4d23a63b-7c61-46af-b53d-9bb5d2650fab | Organization Fee Schedule |       1 | 2018-12-03 22:39:10.466736 | 2018-12-03 22:39:10.466736

select fsr.* from fee_schedule_ranges fsr join fee_schedules f on f.id = fsr.fee_schedule_id where f.name = 'Organization Fee Schedule' and f.version = 1;
                 id                  |           fee_schedule_id            | min_price | fee_in_cents |         created_at         |         updated_at         
--------------------------------------+--------------------------------------+-----------+--------------+----------------------------+----------------------------
0de49c41-7866-4e77-b3f9-f857d6fad56a | 4d23a63b-7c61-46af-b53d-9bb5d2650fab |       100 |           50 | 2018-12-03 22:39:10.466736 | 2018-12-03 22:39:10.466736
```

Create new fee schedule with ranges from original fee schedule:

```
curl http://localhost:9000/organizations/39446110-6f19-4f81-8002-b526b74beb2e/fee_schedule --header "Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIwMGI0MThiMC0yYzg0LTQ5NjktODliZi0xNTQ4ZWRhNzVhMmEiLCJpc3MiOiJiZy1zYW1wbGUtaXNzdWVyIiwiZXhwIjoxNTQzODc3NTQzfQ.JigOQwrkeXCwmMNPswQjMA0zqt8hOt9QCrXN4D0-8Ck" --request POST --data '{"name": "Organization Fee Schedule", "ranges": [{"min_price": 100, "fee_in_cents": 50}, {"min_price": 1000, "fee_in_cents": 150}]}' --header "Content-Type: application/json" -i
{"id":"c8439849-189c-4342-a3d3-562549dc3448","name":"Organization Fee Schedule","version":2,"created_at":"2018-12-03T22:39:55.028374","ranges":[{"id":"0ab857e1-e05e-4b03-89d0-b80da2bb9092","fee_schedule_id":"c8439849-189c-4342-a3d3-562549dc3448","min_price":100,"fee_in_cents":50,"created_at":"2018-12-03T22:39:55.028374","updated_at":"2018-12-03T22:39:55.028374"},{"id":"d1b16f0c-1e6f-4edf-be7c-191ba9a2f686","fee_schedule_id":"c8439849-189c-4342-a3d3-562549dc3448","min_price":1000,"fee_in_cents":150,"created_at":"2018-12-03T22:39:55.028374","updated_at":"2018-12-03T22:39:55.028374"}]}

select * from fee_schedules where name = 'Organization Fee Schedule';
                id                  |           name            | version |         created_at         |         updated_at         
--------------------------------------+---------------------------+---------+----------------------------+----------------------------
02847fb9-bb0b-4ec6-81f9-2da091df484c | Organization Fee Schedule |       0 | 2018-12-03 22:38:03.179941 | 2018-12-03 22:38:03.179941
4d23a63b-7c61-46af-b53d-9bb5d2650fab | Organization Fee Schedule |       1 | 2018-12-03 22:39:10.466736 | 2018-12-03 22:39:10.466736
c8439849-189c-4342-a3d3-562549dc3448 | Organization Fee Schedule |       2 | 2018-12-03 22:39:55.028374 | 2018-12-03 22:39:55.028374

select fsr.* from fee_schedule_ranges fsr join fee_schedules f on f.id = fsr.fee_schedule_id where f.name = 'Organization Fee Schedule' and f.version = 2;
                 id                  |           fee_schedule_id            | min_price | fee_in_cents |         created_at         |         updated_at         
--------------------------------------+--------------------------------------+-----------+--------------+----------------------------+----------------------------
0ab857e1-e05e-4b03-89d0-b80da2bb9092 | c8439849-189c-4342-a3d3-562549dc3448 |       100 |           50 | 2018-12-03 22:39:55.028374 | 2018-12-03 22:39:55.028374
d1b16f0c-1e6f-4edf-be7c-191ba9a2f686 | c8439849-189c-4342-a3d3-562549dc3448 |      1000 |          150 | 2018-12-03 22:39:55.028374 | 2018-12-03 22:39:55.028374
```

**Comment by Krakaw** [comment]
@Timo614 This will be closed by #639 as we need to add some new fields to it anyway.

**Comment by Krakaw** [comment]
The issue was it was being sorted by the uuid instead of the version. We can close this issue, there are other issues with the fee schedules we'll need to fix anyway.
